core/src/main/java/nl/nn/adapterframework/receivers/Receiver.java

```diff
@@ -902,17 +918,18 @@ private void moveInProcessToError(String originalMessageId, String correlationId
 			if (errorSender!=null) {
 				errorSender.sendMessage(message, null);
 			}
-			ProcessState targetState = null;
-			if (knownProcessStates.contains(ProcessState.ERROR)) {
-				targetState = ProcessState.ERROR;
-			} else {
-				if (knownProcessStates.contains(ProcessState.DONE)) {
-					targetState = ProcessState.DONE;
-				}
-			}
-			if (targetState !=null && getListener() instanceof IHasProcessState) {
-				changeProcessState(rawMessage, targetState, null);
-			}
+			// processState change is currently handled in listener.afterMessageProcessed()
+//			ProcessState targetState = null;
+//			if (knownProcessStates.contains(ProcessState.ERROR)) {
+//				targetState = ProcessState.ERROR;
+//			} else {
+//				if (knownProcessStates.contains(ProcessState.DONE)) {
+//					targetState = ProcessState.DONE;
+//				}
+//			}
+//			if (targetState !=null && getListener() instanceof IHasProcessState) {
+//				changeProcessState(rawMessage, targetState);
+//			}
```

Comment:
Wat als hij in `hasProblematicHistory(...)` terecht komt? Roept hij dan ook listener.afterMessageProcessed() aan?

Reply:
hasProblematicHistory() roept  moveInProcessToErrorAndDoPostProcessing() aan, en die eindigt met afterMessageProcessed().